### PR TITLE
Add Thanos Receiver

### PR DESCRIPTION
**Pull request by varet80**
Adding support to install Thanos Receiver. it is an alternative to sidecar.
Recevier can replace partially sidecar and allow remote_write

### CODE REVIEW

File: charts/thanos/templates/receive/service-headless.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  type: ClusterIP
+  clusterIP: None
+  ports:
+    - name: grpc
+      port: 10907
+      targetPort: grpc
+    - name: http-remote-write
+      protocol: TCP
+      port: 10908
+      targetPort: remote-write
```
Comment (stevehipwell) [inline charts/thanos/templates/receive/service-headless.yaml:20]:
```suggestion
      targetPort: http-remote-write
```
I'd prefer the port names to match. Also is there a reason why haven't you used the same port schema as the rest of the chart?

File: charts/thanos/values.yaml [hunk 2]
```diff
@@ -283,6 +283,87 @@ queryFrontend:
...
+
+  terminationGracePeriodSeconds:
+
+  extraEnv: []
+
+  extraArgs: []
+  receiveAppLabels:
+    - name: "location"
+      value: "main"
+  tsdbRetention: "15d"
```
Comment (stevehipwell) [inline charts/thanos/values.yaml:328]: Based on the other chart components strongly typed arguments live above `extraArgs`. Also are these flags required and are these appropriate values?

**Review by stevehipwell — CHANGES_REQUESTED** [review]
Thanks for the PR @varet80, this was on my radar but I hadn't got around to it yet. I've added some questions suggestions.

File: charts/thanos/values.yaml (hunk 2 above)
Comment (varet80) [reply charts/thanos/values.yaml:328]: This is mistake! I will remove these. as I was playing around

Comment (varet80) [reply charts/thanos/values.yaml:328]: The `receiveAppLabel` is used for tsdb db to separate databases
stated on documentation:
```
      --label=key="value" ...    External labels to announce. This flag will be
                                 removed in the future when handling multiple
                                 tsdb instances is added.
```
retention: https://thanos.io/tip/components/receive.md/#tenant-lifecycle-manage

with any of these missing, as stated. receiver do not start.

Comment (stevehipwell) [reply charts/thanos/values.yaml:328]: I'm not sold on the labelling logic being deigned like this; have you looked at what they do for labels in [kube-thanos](https://github.com/thanos-io/kube-thanos)? I think `--label=receive="true"` & `--label=replica="$(NAME)"` would be the best defaults with end-users able to pass in any additional label arguments they wanted. Maybe with an extra value to disable default labels from being created to allow full control.

I also think `15d` for the default retention seems too high; I don't know what the cost of recreating a tenant is but I suspect it's likely low given that all data is pushed to object storage after `2h`. I'd be thinking something closer to `4h` or `6h` for the default.

File: charts/thanos/templates/receive/service-headless.yaml (hunk 1 above)
Comment (stevehipwell) [reply charts/thanos/templates/receive/service-headless.yaml]: I don't think this has been resolved.

File: charts/thanos/values.yaml (hunk 2 above)
Comment (varet80) [reply charts/thanos/values.yaml:328]: The label in thanos distinguishes between multiple receivers. 
And they require them to be set. (Datacenters etc)

receive="true" is a strange label to set on all your metrics. these are labels added into the stored metrics.

Comment (stevehipwell) [reply charts/thanos/values.yaml:328]: It's been a while, let me re-read the docs for this point.

Comment (varet80) [reply charts/thanos/values.yaml:328]: some supporting reading: https://thanos.io/tip/operating/multi-tenancy.md/

**Comment by stevehipwell** [comment]
@varet80 I think #623 should have introduced the functionality that you require so I'm going to close this PR. Please feel free to open a new one of there is missing functionality that you can't make work with the provided chart configuration.

**Pull request closed without merging**